app/packages/core/src/components/ColorModal/LabelTag.tsx

```diff
@@ -193,20 +209,35 @@ const LabelTag: React.FC = () => {
           >
             {/* set attribute value - color */}
             <Checkbox
-              name={`Use custom colors for specific field values`}
+              name={`Use custom colors for specific label tag values`}
               value={state.useLabelColors}
               setValue={(v: boolean) => {
-                const copy = cloneDeep(labelTags);
-                copy.valueColors = v
-                  ? [{ value: "", color: defaultColor }]
-                  : [];
-
-                setColorScheme({
-                  colorPool,
-                  fields,
-                  labelTags: copy,
+                setSetting((cur) => {
+                  if (!cur) {
+                    cur = { valueColors: [] };
+                  }
+
+                  if (!cur?.valueColors?.length && v) {
+                    cur = {
+                      ...cur,
+                      valueColors: [
+                        {
+                          value: "",
+                          color:
+                            colorPool[
+                              Math.floor(Math.random() * colorPool.length)
+                            ],
+                        },
+                      ],
+                    };
+                  } else if (!v) {
+                    cur = { ...cur, valueColors: [] };
+                  }
+
+                  return {
+                    ...cur,
+                  };
```

Comment:
nit: I think this can be simplified 
```
setSetting((cur) => {
  if (!cur?.valueColors?.length && v) {
    return {
      ...cur,
      valueColors: [{
        value: "",
        color: colorPool[Math.floor(Math.random() * colorPool.length)],
      }],
    };
  } else if (!v) {
    return { ...cur, valueColors: [] };
  }

  return { ...cur };
});
```